pkg/beholder/message.go

```diff
@@ -0,0 +1,275 @@
+
+func (a Attributes) Add(attrKVs ...any) Attributes {
+	l := len(attrKVs)
+	for i := 0; i < l; {
+		switch t := attrKVs[i].(type) {
+		case map[string]any:
+			for k, v := range t {
+				a[k] = v
+			}
+			i++
+		case Attributes:
+			for k, v := range t {
+				a[k] = v
+			}
+			i++
+		case string:
+			if i+1 >= l {
+				break
+			}
+			val := attrKVs[i+1]
+			a[t] = val
+			i += 2
+		default:
+			// Unexpected type
+			return a
+		}
+	}
+	return a
+}
+
+func NewMessage(body []byte, attrKVs ...any) Message {
+	return Message{
+		Body:  body,
+		Attrs: NewAttributes(attrKVs...),
+	}
+}
+
+func (e *Message) AddAttributes(attrs Attributes) {
+	if e.Attrs == nil {
+		e.Attrs = make(map[string]any, len(attrs))
+	}
+	for k, v := range attrs {
+		e.Attrs[k] = v
+	}
+}
+
+func (e *Message) AddOtelAttributes(attrs ...attribute.KeyValue) {
+	if e.Attrs == nil {
+		e.Attrs = make(map[string]any, len(attrs))
+	}
+	for _, v := range attrs {
+		e.Attrs[string(v.Key)] = v.Value
+	}
+}
+
+func (e *Message) OtelRecord() otellog.Record {
+	return newRecord(e.Body, e.Attrs)
+}
+
+func (e *Message) SdkOtelRecord() otelsdklog.Record {
+	return newSdkRecord(e.Body, e.Attrs)
+}
+
+func (e *Message) Copy() Message {
+	attrs := make(Attributes, len(e.Attrs))
+	for k, v := range e.Attrs {
+		attrs[k] = v
+	}
+	c := Message{
+		Attrs: attrs,
+	}
+	if e.Body != nil {
+		c.Body = make([]byte, len(e.Body))
+		copy(c.Body, e.Body)
+	}
+	return c
+}
+
+// Creates otellog.Record from body and attributes
+func newRecord(body []byte, attrs map[string]any) otellog.Record {
+	otelRecord := otellog.Record{}
+	if body != nil {
+		otelRecord.SetBody(otellog.BytesValue(body))
+	}
+	for k, v := range attrs {
+		otelRecord.AddAttributes(OtelAttr(k, v))
+	}
+	return otelRecord
+}
+
+// Creates otelsdklog.Record from body and attributes
+// NOTE: internal function otelsdklog.newRecord returns value not pointer
+func newSdkRecord(body []byte, attrs map[string]any) otelsdklog.Record {
+	sdkRecord := otelsdklog.Record{}
+	if body != nil {
+		sdkRecord.SetBody(otellog.BytesValue(body))
+	}
+	for k, v := range attrs {
+		sdkRecord.AddAttributes(OtelAttr(k, v))
+	}
+	return sdkRecord
+}
+
+func OtelAttr(key string, value any) otellog.KeyValue {
+	switch v := value.(type) {
+	case string:
+		return otellog.String(key, v)
+	case []string:
+		vals := make([]otellog.Value, 0, len(v))
+		for _, s := range v {
+			vals = append(vals, otellog.StringValue(s))
+		}
+		return otellog.Slice(key, vals...)
+	case int64:
+		return otellog.Int64(key, v)
+	case int:
+		return otellog.Int(key, v)
+	case float64:
+		return otellog.Float64(key, v)
+	case bool:
+		return otellog.Bool(key, v)
+	case []byte:
+		return otellog.Bytes(key, v)
+	case nil:
+		return otellog.Empty(key)
+	case otellog.Value:
+		return otellog.KeyValue{Key: key, Value: v}
+	case attribute.Value:
+		return OtelAttr(key, v.AsInterface())
+	default:
+		return otellog.String(key, fmt.Sprintf("<unhandled beholder attribute value type: %T>", v))
```

Comment:
Should we try to print the value too? e.g. with `%v`?

Reply:
@jmank88 Addressed https://github.com/smartcontractkit/chainlink-common/pull/694/commits/84357e5185442a17dd870226c45b83cd983dbd7b